drivers/adc/ad463x/ad463x.c

```diff
@@ -306,42 +316,72 @@ int32_t ad463x_set_ch_offset(struct ad463x_dev *dev, uint8_t ch_idx,
 /**
  * @brief Initialize GPIO driver handlers for the GPIOs in the system.
  *        ad463x_init() helper function.
- * @param [out] dev - ad463x_dev device handler.
- * @param [in] init_param - Pointer to the initialization structure.
- * @return 0 in case of success, -1 otherwise.
+ * @param dev - ad463x_dev device handler.
+ * @param init_param - Pointer to the initialization structure.
+ * @return 0 in case of success, negative number otherwise.
  */
 static int32_t ad463x_init_gpio(struct ad463x_dev *dev,
 				struct ad463x_init_param *init_param)
 {
 	int32_t ret;
-
+	/* configure reset pin */
 	ret = no_os_gpio_get_optional(&dev->gpio_resetn, init_param->gpio_resetn);
 	if (ret != 0)
-		return ret;
+		goto error_gpio_reset;
 
 	/** Reset to configure pins */
 	if (dev->gpio_resetn) {
 		ret = no_os_gpio_direction_output(dev->gpio_resetn, NO_OS_GPIO_LOW);
 		if (ret != 0)
-			return ret;
+			goto error_gpio_reset;
 
 		no_os_mdelay(100);
 		ret = no_os_gpio_set_value(dev->gpio_resetn, NO_OS_GPIO_HIGH);
 		if (ret != 0)
-			return ret;
+			goto error_gpio_reset;
 
 		no_os_mdelay(100);
 	}
 
+	/** Configure PGIA pins, if available */
+	if (dev->has_pgia) {
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a0,init_param->gpio_pgia_a0);
+		if (ret != 0)
+			goto error_gpio_pgia;
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a1, init_param->gpio_pgia_a1);
+		if (ret != 0)
+			goto error_gpio_pgia;
+		/* set default values */
+		if (dev->gpio_pgia_a0) {
```

Comment:
I don't think these checks are needed, the goto statement will occur if dev->gpio_pgia_a0 is null, the same is valid below